### ISSUE TITLE
Improve the debug logging output

### ISSUE THREAD

**Issue by luziferius** [autokey/autokey#230]
## Classification:

Enhancement

## Summary

There are several outdated, wrong, or oddly worded logging messages.
The task is to collect all those wrong, odd or spammy logging messages and improve them.
Find largely unlogged code and add some logging messages (but not into hot code paths).

### Wrong messages
- [x] The logging seemingly *always* says `DEBUG - service - No phrase/script matched hotkey`
Add a message if an item is triggered by an abbreviation to make the logic flow more obvious. Then each block of action starts with either a matched hotkey or a matched abbreviation. 
- [x] Odd wording, should say *Phrase*, not abbreviation, as that confuses terminology:
https://github.com/autokey/autokey/blob/b875c2690f45a8776a3bedd3c3ae525bd6e6f5f2/lib/autokey/service.py#L415
- [x] This is a queue, not a stack… `DEBUG - service - Input stack at end of handle_keypress: deque([' '], maxlen=150)`
- [x] At startup, AutoKey seems to dump the key map into the log, like `2019-02-10 12:02:52,443 DEBUG - interface - [N]: [(57, 1), (57, 3)]`. These messages don’t look like they are that useful. But they spam the log. So remove the log message.
- [x] During startup, this useless line is logged: `2019-02-11 07:14:48,749 DEBUG - interface - <map object at 0x7f612e04a7b8>
- [ ] Use clear wording when anything is triggered by an Abbreviation or Hotkey. See https://github.com/autokey/autokey/issues/338#issuecomment-568624841
- [ ] Log the window filter that matched the window title or WM_CLASS.
- [ ] Reduce the frequency of dumping the input queue. Maybe dump it every 10 keystrokes? Maybe this is not needed, if the normal INFO level logging is sufficient.

## Version

AutoKey version: git master

Used GUI (Gtk, Qt, or both): both

Installed via: local git checkout

**Comment by Elliria** [comment]
I've added part of this to my plate.

**Comment by josephj11** [comment]
If possible, it woud be great to add an initial log message displaying the front end (Qt or GTK) and the version of AutoKey. This would ideally be done so it happens only once, not repeatedly with events.

Since we often get snips of the end of a log, it might also be good to include this when AutoKey closes with tracing enabled.

Other information might also be included. Like whether X11 or Wayland is running.

**Comment by Elliria** [comment]
Both are excellent ideas and shouldn't be hard to implement.

**Comment by josephj11** [comment]
If you add that info, it might be good to add it as a new function (internal - not an API call ...) so the code can be reused if we want to call it somewhere else in the code later.

Might be cool to add it somewhere like as a CLI flag similar to --version so it could be accessed without running a trace. --info ??? (nice but not a priority)

**Comment by Elliria** [comment]
Maybe **-e** and **--env** would be good, although adding options is something I'd still need to learn how to do (finding all the places where references to them would need to go, etc.).

**Comment by josephj11** [comment]
@Elliria those flags sound better, but adding that functionality can be split out from this enhancement and added separately later. Building the code as a function just makes it easier to do that later.

**Comment by Elliria** [comment]
Yep. The options would just be a way to use the underlying code.

**Comment by josephj11** [comment]
I just thought of another "blue sky" use. It could somehow  be returned as a string that the user could use, e.g. to display in a dialog box.  Not sure how useful that would be, but it's another possibility.

**Comment by Elliria** [comment]
That one probably wouldn't be necessary since we can easily script that ourselves by putting the result of the command that checks if Wayland is running into a dialog. We might, however, be able to make it part of the information displayed in the **Help** menu as the **About environment** entry (or something like that) to go along with the existing **About AutoKey** entry in a similar way as **Dolphin** offers the **About Dolphin** and **About KDE** entries in the **Help** menu, as all other Qt programs do.

**Comment by josephj11** [comment]
You can script it yourself if you code the whole thing from scratch or use a new API call that we would have to add.

We already have About AutoKey and About Qt/GTK. It would have to be wedged in there somehow. It sort-of fits in either of those, but not really.

**Comment by Elliria** [comment]
The **About environment** entry would get around the issue of it not quite fitting in with either of those entries.

**Comment by josephj11** [comment]
OK. You can open an enhancement issue for it if you want to.

**Comment by Elliria** [comment]
I will at some point. It's on my local ToDo list.

**Comment by josephj11** [comment]
If this ever gets worked on again, it would be nice to add at least two logging levels. There are some messages that repeat a lot like capturing key events after a fake keypress call that are usually useless, but might be needed in some specialized circumstances.

It would be nice to do a logrotate on `$HOME/.local/share/autokey/autokey.log` which just grows forever now even though almost no one ever looks at it.

**Comment by dlk3** [comment]
> It would be nice to do a logrotate on `$HOME/.local/share/autokey/autokey.log` which just grows forever now even though almost no one ever looks at it.

AutoKey already includes code in ```lib/autokey/logger.py``` which rotates ```$HOME/.local/share/autokey/autokey.log``` when it reaches 5MB in size.  Three backup logs are kept in rotation.  That limits the maximum total log file space usage to 20MB.

**Comment by josephj11** [comment]
@dlk3 Cool! That one slipped by me. I thought my log was a bit smaller than expected.
